### PR TITLE
Split universal adapter

### PR DESCRIPTION
**Pull request by jacek-marchwicki**

### CODE REVIEW

File: example-model/build.gradle [hunk 1]
```diff
@@ -11,6 +11,9 @@ dependencies {
     compile "com.google.code.gson:gson:2.4"
     compile "org.json:json:20131018"
 
+    // Recycler view helpers
+    compile 'com.github.jacek-marchwicki.recyclerview-changes-detector:universal-adapter-java:-SNAPSHOT'
```
Comment (jacek-marchwicki) [inline example-model/build.gradle]: Use version instead of snapshot

File: example/build.gradle [hunk 2]
```diff
@@ -15,6 +15,9 @@ dependencies {
     compile 'com.android.support:recyclerview-v7:24.2.1'
     compile 'com.android.support:cardview-v7:24.2.1'
 
+    // Recycler view helpers
+    compile 'com.github.jacek-marchwicki.recyclerview-changes-detector:universal-adapter-rx:-SNAPSHOT'
```
Comment (jacek-marchwicki) [inline example/build.gradle]: Use version instead of snapshot

**Comment by m-zagorski** [comment]
I would even go further and instead of `deprecated` removed the classes totally.
There starts to be some mess due to all those deprecated things, or do this deprecated now and release some major new version without all deprecated methods and classes, what do you think?

**Comment by jacek-marchwicki** [comment]
@m-zagorski we can do this.. but it could be pain in the ass if we remove it.

I.e. we use old UniversalAdapter in `libraryA` (`libraryA` depends of `rx-java-extensions:1.0.0`).
Your project `projectB` depends on `libraryA`.
Your project `projectB` use also UniversalAdapter  (`projectB` depends of `rx-java-extensions:1.0.0`).
So when building `projectB` you will use `rx-java-extensions:1.0.0`.

If you switch to new UniversalAdapter and change requirements of `libraryA` to `rx-java-extensions:2.0.0`. Then update `projectB` to use newer version of `projectA` You app will crash at runtime because whole project will use `rx-java-extensions:2.0.0` and will crash because `projectB` use UniversalAdapter that was removed from jar.

The same can happen if you decide to update `projectB` to `rx-java-extensions:2.0.0` without updating `libraryA`.

So I think we should keep old methods for some time.

**Comment by jacek-marchwicki** [comment]
cc: @krpiotrek

**Pull request merged**